iceoryx_utils/test/moduletests/test_cxx_list.cpp

```diff
@@ -0,0 +1,2283 @@
+    ASSERT_THAT(cTor, Eq(TESTLISTCAPACITY));
+    ASSERT_THAT(dTor, Eq(TESTLISTCAPACITY));
+}
+
+TEST_F(list_test, PopBackFullPlusOneToEmptyList)
+{
+    // fill even more than size
+    for (uint64_t i = 0; i < TESTLISTCAPACITY; ++i)
+    {
+        sut.emplace(sut.cbegin());
+        EXPECT_THAT(sut.size(), Eq((i + 1) > TESTLISTCAPACITY ? TESTLISTCAPACITY : (i + 1)));
+    }
+
+    for (uint64_t i = 0; i < TESTLISTCAPACITY; ++i)
+    {
+        EXPECT_THAT(sut.size(), Eq(TESTLISTCAPACITY - i));
+        EXPECT_TRUE(sut.pop_back());
+    }
+
+    EXPECT_FALSE(sut.pop_back());
+
+    ASSERT_THAT(sut.size(), Eq(0u));
+    ASSERT_THAT(cTor, Eq(TESTLISTCAPACITY));
+    ASSERT_THAT(customCTor, Eq(0u));
+    ASSERT_THAT(dTor, Eq(TESTLISTCAPACITY));
+}
+
+
+TEST_F(list_test, InsertEmptyListAsLValue)
+{
+    constexpr int64_t DEFAULT_VALUE{13};
+    const TestListElement a{DEFAULT_VALUE};
+
+    sut.insert(sut.cbegin(), a);
+
+    ASSERT_THAT(sut.size(), Eq(1u));
+    ASSERT_THAT(cTor, Eq(0u));
+    ASSERT_THAT(customCTor, Eq(1u));
+    ASSERT_THAT(copyCTor, Eq(1u));
+    ASSERT_THAT(moveCTor, Eq(0u));
+    ASSERT_THAT(copyAssignment, Eq(0u));
+    ASSERT_THAT(moveAssignment, Eq(0u));
+}
+
+TEST_F(list_test, InsertLValueCheckReturn)
+{
+    constexpr int64_t DEFAULT_VALUE{13};
+    const TestListElement a{DEFAULT_VALUE};
+
+    auto iter = sut.insert(sut.begin(), a);
+
+    ASSERT_THAT(iter == sut.begin(), Eq(true));
+}
+
+TEST_F(list_test, InsertEmptyListAsRValue)
+{
+    sut.insert(sut.begin(), {10});
+
+    ASSERT_THAT(sut.size(), Eq(1u));
+    ASSERT_THAT(cTor, Eq(0u));
+    ASSERT_THAT(customCTor, Eq(1u));
+    ASSERT_THAT(copyCTor, Eq(0u));
+    ASSERT_THAT(moveCTor, Eq(1u));
+    ASSERT_THAT(copyAssignment, Eq(0u));
+    ASSERT_THAT(moveAssignment, Eq(0u));
+}
+
+TEST_F(list_test, InsertRValueCheckReturn)
+{
+    auto iter = sut.insert(sut.begin(), {10});
+
+    ASSERT_THAT(iter == sut.begin(), Eq(true));
+    ASSERT_THAT((*iter).m_value, Eq(10));
+}
+
+TEST_F(list_test, InsertBeginListLValue)
+{
+    constexpr int64_t DEFAULT_VALUE{13};
+    const TestListElement a{DEFAULT_VALUE};
+
+    sut.insert(sut.begin(), a);
+    sut.emplace_front();
+
+    ASSERT_THAT(sut.size(), Eq(2u));
+    ASSERT_THAT(cTor, Eq(1u));
+    ASSERT_THAT(customCTor, Eq(1u));
+    auto iter = sut.begin();
+    EXPECT_THAT(iter->m_value, Eq(TEST_LIST_ELEMENT_DEFAULT_VALUE));
+    EXPECT_THAT((++iter)->m_value, Eq(DEFAULT_VALUE));
+}
+
+
+TEST_F(list_test, InsertBeforeBeginListLValue)
+{
+    constexpr int64_t DEFAULT_VALUE{13};
+    const TestListElement a{DEFAULT_VALUE};
+
+    sut.emplace_front();
+    sut.insert(sut.begin(), a);
+
+    ASSERT_THAT(sut.size(), Eq(2u));
+    ASSERT_THAT(cTor, Eq(1u));
+    ASSERT_THAT(customCTor, Eq(1u));
+    auto iter = sut.begin();
+    EXPECT_THAT((*iter).m_value, Eq(DEFAULT_VALUE));
+    EXPECT_THAT((++iter)->m_value, Eq(TEST_LIST_ELEMENT_DEFAULT_VALUE));
+}
+
+TEST_F(list_test, InsertBeforeBeginListRValue)
```

Comment:
I am thinking about how one could have reused much the code for test cases when the difference is just an lvalue or an rvalue. Probably with some kind of policy and typed tests, where the policy type is varied and provides a generation interface the allows us to obtain rvalues or lvalues (since returned values are always rvalues there would be some tricks needed which seem ugly).

Summary: a strategy write tests in a more generic way suitable for for lvalues and  rvalues would be nice but it is unclear how this could be done elegantly. Food for thought for the future. 
No change needed, of course.